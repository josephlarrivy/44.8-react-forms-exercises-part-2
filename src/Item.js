import React from "react";

const Item = ({ id, item, removeItem }) => {

    const handleClick = evt => {
        const id =  evt.target.parentNode.id;
        removeItem(id)
        
    }

    return (
        <>
            <li id={id}>{item}  <button onClick={handleClick}>  remove</button></li>
        </>
    )

}

export default Item;


