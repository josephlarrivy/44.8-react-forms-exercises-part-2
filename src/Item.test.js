import { render, screen } from '@testing-library/react';
import Item from './Item'

test('renders an item', () => {
    render(<Item />);
});

it("matches snapshot", function () {
    const { asFragment } = render(<Item id='testingId1' key='testingId1' item='testingItem1'/>);
    expect(asFragment()).toMatchSnapshot();
});