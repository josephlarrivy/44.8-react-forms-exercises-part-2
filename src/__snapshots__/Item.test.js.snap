// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <li
    id="testingId1"
  >
    testingItem1  
    <button>
        remove
    </button>
  </li>
</DocumentFragment>
`;
